File: src/main/java/fastcampus/innercircle/onboarding/survey/domain/SurveyResponse.java
package fastcampus.innercircle.onboarding.survey.domain;

import jakarta.persistence.*;

import java.util.List;

@Entity
@Table(name = "SURVEY_RESPONSE")
public class SurveyResponse {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "RESPONSE_ID")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumns({
            @JoinColumn(name = "FORM_ID", referencedColumnName = "FORM_ID"),
            @JoinColumn(name = "VERSION", referencedColumnName = "VERSION")
    })
    private SurveyForm form;

    @OneToMany(mappedBy = "response")
    private List<SurveyResponseDetail> details;
}

File: src/main/kotlin/org/inner/circle/survey/domain/SurveyResponse.kt
package org.inner.circle.survey.domain

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = "survey_response")
class SurveyResponse(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_id", nullable = false)
    var survey: Survey,
    @Column(nullable = false)
    var respondent: String,
) : BaseEntity() {
    constructor() : this(survey = Survey(), respondent = "")
}
